pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -256,108 +296,48 @@ def get_queryset(self):
             # and it fails when "path" is not on self.kwargs
             return Collection.objects.none()
         repo_version = self._repository_version
-        return Collection.objects.filter(
-            versions__in=repo_version.content,
-        )
-
-    def append_context(self, queryset):
-        """Appending collection data to context."""
-        repo_version = self._repository_version
-        collections_qs = queryset.annotate(available_versions=ArrayAgg("versions__version"))
-        versions_context = {}
-        for collection_id, available_versions in collections_qs.values_list(
-            "pk", "available_versions"
-        ):
-            versions_context[collection_id] = available_versions
-
-        self.available_versions_context = versions_context  # needed by get__serializer_context
-        self._deprecation
-
-        collections = Collection.objects.filter(
-            pk__in=versions_context.keys(),
-            versions__version_memberships__repository=repo_version.repository,
-        ).annotate(
-            repo_version_added_at=Window(
-                expression=FirstValue(
-                    "versions__version_memberships__version_added__pulp_last_updated"
-                ),
-                partition_by=[F("versions__collection_id")],
-                order_by=F("versions__version_memberships__version_added__pulp_last_updated").desc(
-                    nulls_last=True
-                ),
-            ),
-            repo_version_removed_at=Window(
-                expression=FirstValue(
-                    "versions__version_memberships__version_removed__pulp_last_updated"
-                ),
-                partition_by=[F("versions__collection_id")],
-                order_by=F(
-                    "versions__version_memberships__version_removed__pulp_last_updated"
-                ).desc(nulls_last=True),
-            ),
-        )
 
-        return collections.distinct("versions__collection_id").only(
-            "pulp_created", "name", "namespace"
+        deprecated_qs = filter_content_for_repo_version(
+            AnsibleCollectionDeprecated.objects, repo_version
+        ).filter(namespace=OuterRef("namespace"), name=OuterRef("name"))
+
+        latest_cv_version_qs = (
+            filter_content_for_repo_version(CollectionVersion.objects, repo_version)
```

Comment:
Have you try using `distinct("namespace", "name")` on this query after the `order_by` to see if it speeds it up?

Reply:
I don't think that would matter. It's already filtering by `.filter(collection=OuterRef("pk"))`